src/backend/distributed/commands/foreign_server.c

```diff
@@ -41,7 +42,14 @@ PreprocessCreateForeignServerStmt(Node *node, const char *queryString,
 		return NIL;
 	}
 
+	/* check creation against multi-statement transaction policy */
+	if (!ShouldPropagateCreateInCurrentTransaction())
+	{
+		return NIL;
+	}
+
 	EnsureCoordinator();
+	EnsureSequentialMode(OBJECT_FOREIGN_SERVER);
```

Comment:
What is the criteria for this check to be necessary? 
For example, it is done for foreign server and extension creation, but not for collation creation.

Reply:
It should be in any node scoped object, if it is still missing from collations it is an oversight and I will add it there.
Not directly related, but it stood out to me while I was going through this code to add the check introduced in this PR.

When we don't enforce sequential mode we get visibility problems if, in this case, the foreign server gets used in a distributed table when both are created in the same transaction. This ensure will transparently switch to sequential to make sure that if we directly propagate the object we will see the object in subsequent statements.